environment.h

```diff
@@ -34,7 +42,8 @@ class environment
 public:
     ///deprecated(sort of)
     environment(double target_valueA, double target_valueB,
-                std::function<double(std::vector<double>)> env_functionA = &env_func_A);
+                std::function<double(std::vector<double>)> env_functionA = &env_func_1,
+                std::function<double(std::vector<double>)> env_functionB = &env_func_2);
```

Comment:
mmm, maybe be consistent across `env_param `and `environment`? this could cause confusion later on.
choose if you want letters or numbers

Reply:
I had switched to numbers because I figured this is two different things: 
the ones in environment are always A and B, 
but depending on how they get initialized we could say that A is going to be function 4 and B is going to be function 2 
so we can pick and choose which functions we use when simulations
Does that make sense?

Reply:
ahnnnnnn!!!! Seems a nice idea but... _future creep_ might be playing you a trick...

Reply:
I see your point anyway, but then in #65 you are using different naming again I think. Be careful. Your call anyway how to call them.

Reply:
Yes I changed back to not have to change your test since I wasn't sure if that makes sense, I will fix and make everything consistent everywhere, and I think I like the numbers so i'll keep that